WordPress/src/main/java/org/wordpress/android/ui/accounts/LoginMagicLinkInterceptActivity.java

```diff
@@ -15,21 +16,55 @@
  * or signup based on deep link scheme, host, and parameters.
  */
 public class LoginMagicLinkInterceptActivity extends Activity {
+    private String mAction;
+    private Uri mUri;
+
     @Inject protected LoginAnalyticsListener mLoginAnalyticsListener;
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
         ((WordPress) getApplication()).component().inject(this);
 
-        mLoginAnalyticsListener.trackLoginMagicLinkOpened();
+        mAction = getIntent().getAction();
+        mUri = getIntent().getData();
 
         Intent intent = new Intent(this, WPMainActivity.class);
-        intent.setAction(getIntent().getAction());
-        intent.setData(getIntent().getData());
+        intent.setAction(mAction);
+        intent.setData(mUri);
+
+        if (hasMagicLinkLoginIntent()) {
+            intent.putExtra(WPMainActivity.ARG_IS_MAGIC_LINK_LOGIN, true);
+
+            if (hasMagicLinkSignupIntent()) {
+                mLoginAnalyticsListener.trackSignupMagicLinkOpened();
+                intent.putExtra(WPMainActivity.ARG_IS_MAGIC_LINK_SIGNUP, true);
+            } else {
+                mLoginAnalyticsListener.trackLoginMagicLinkOpened();
+                intent.putExtra(WPMainActivity.ARG_IS_MAGIC_LINK_SIGNUP, false);
+            }
+        }
+
         intent.setFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP | Intent.FLAG_ACTIVITY_NEW_TASK
                         | Intent.FLAG_ACTIVITY_CLEAR_TASK);
         startActivity(intent);
         finish();
     }
+
+    private boolean hasMagicLinkLoginIntent() {
+        String host = (mUri != null && mUri.getHost() != null) ? mUri.getHost() : "";
+        return Intent.ACTION_VIEW.equals(mAction) && host.contains(LoginActivity.MAGIC_LOGIN);
+    }
+
+    private boolean hasMagicLinkSignupIntent() {
+        if (mUri != null) {
+            String parameter = SignupEpilogueActivity.MAGIC_SIGNUP_PARAMETER;
+            String value = (mUri.getQueryParameterNames() != null && mUri.getQueryParameter(parameter) != null)
+                    ? mUri.getQueryParameter(parameter) : "";
+            return Intent.ACTION_VIEW.equals(mAction) && mUri.getQueryParameterNames().contains(parameter)
```

Comment:
I am not really sure why we need the `mUri.getQueryParameterNames().contains(parameter)` since we already check for the value afterwards. This can also create an NPE.

Reply:
Looks like this code was only moved. I still feel like we should address it here if we can.

Reply:
I removed the extra condition in fd441c8.